lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,165 @@
+                           type: nil,
+                           message_id: :auto_generate,
+                           headers: nil)
+      if delayed
+        new.send_delayed_message!(*[payload, routing_key, delay_by].compact, type: type, headers: headers, message_id: message_id)
+      else
+        new.send_message!(payload, routing_key, type: type, headers: headers, message_id: message_id)
+      end
+      logf "AFTER Transmitting Message on %{routing_key} -> %{payload}",routing_key: routing_key, payload: payload
+      true
+    rescue => e
+
+      logf "ERROR Transmitting Message on %{routing_key} -> %{payload} : %{error}", routing_key: routing_key, payload: payload, error: e, at: :error
+
+      case on_error
+
+        when :raise
+          raise e
+
+        when :resque, :retry_async
+          begin
+            send_message_async(payload, routing_key, delay_by_ms: delayed ? delay_by || DEFAULT_DELAY_BY_MS : 0)
+          rescue => exception
+            warn(exception.message)
+            raise e
+          end
+
+        else # ignore
+      end
+      false
+    end
+
+    # Enqueue the message with the configured background processor.
+    # - :delay_by_ms:: Integer milliseconds to delay the message. Default is 0.
+    def self.send_message_async(payload, routing_key,
+                                delay_by_ms: 0,
+                                type: nil,
+                                message_id: :auto_generate,
+                                headers: nil)
+      background_job_processor = Pwwka.configuration.background_job_processor
+      job = Pwwka.configuration.async_job_klass
+
+      if background_job_processor == :resque
+        # Be perhaps too carefully making sure we queue jobs in the legacy way
+        if type == nil && message_id == :auto_generate && headers == nil
+          Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key)
+        else
+          Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key, type: type, message_id: message_id, headers: headers)
+        end
+      elsif background_job_processor == :sidekiq
+        options = { delay_by_ms: delay_by_ms, type: type, message_id: message_id, headers: headers }
+        job.perform_async(payload, routing_key, options)
+      end
+    end
+
+    # Send a less important message that doesn't have to go through. This eats
+    # any `StandardError` and logs it, returning false rather than blowing up.
+    #
+    # payload:: Hash of what you'd like to include in your message
+    # routing_key:: String routing key for the message
+    # delayed:: Boolean send this message later
+    # delay_by:: Integer milliseconds to delay the message
+    #
+    # Returns true if the message was sent, false otherwise
+    # @deprecated This is ignoring a message. ::send_message supports this explicitly.
+    def send_message_safely(payload, routing_key, delayed: false, delay_by: nil, message_id: :auto_generate)
```

Comment:
go ahead and remove this method from this new class.  we don't want to make it easy to ignore errors while sending messages